libvast/src/system/spawn_explorer.cpp

```diff
@@ -27,18 +29,66 @@
 #include <caf/local_actor.hpp>
 #include <caf/settings.hpp>
 
+#include <optional>
+
 using namespace std::chrono_literals;
 
 namespace vast::system {
 
+caf::error explorer_validate_args(const caf::settings& args) {
+  auto before_arg = caf::get_if<std::string>(&args, "explore.before");
+  auto after_arg = caf::get_if<std::string>(&args, "explore.after");
+  auto by_arg = caf::get_if(&args, "explore.by");
+  if (!before_arg && !after_arg && !by_arg)
+    return make_error(ec::invalid_configuration, "At least one of '--before', "
+                                                 "'--after' or '--by' must be "
+                                                 "present.");
+  vast::duration before = vast::duration{0s};
+  vast::duration after = vast::duration{0s};
+  if (before_arg) {
+    auto d = to<vast::duration>(*before_arg);
+    if (!d)
+      return make_error(ec::invalid_argument, "Could not parse", *before_arg,
+                        "as duration.");
+    before = *d;
+  }
+  if (after_arg) {
+    auto d = to<vast::duration>(*after_arg);
+    if (!d)
+      return make_error(ec::invalid_argument, "Could not parse", *after_arg,
+                        "as duration.");
+    after = *d;
+  }
```

Comment:
I've tried coming up with a way to make this less repetitive, but every solution involving a lambda still needs the duplicated error handling, which makes me think this is fine.

Reply:
Yeah, I had the same experience, a macro would work but thats not really better :/